src/simulations/collect.jl

```diff
@@ -85,7 +77,8 @@ function collect_agent_data(model::ABM, properties::AbstractArray, step::Int = 0
   dd = DataFrame()
   dd[!, :id] = collect(keys(model.agents))
   for fn in properties
-    dd[!, fn] = get_data.(values(model.agents), fn)
+    colname = typeof(fn) <: Function ? Symbol(fn) : fn
```

Comment:
this `colname` should be made a function that uses multiple dispatch on `::Function` to avoid code duplication.